packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +269,32 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validateIconsPurpose = (icons: ManifestImageResource[], resource: string, location: ProblemLocation | null) => {
+            for (let index = 0; index < icons.length; index++) {
+                const icon = icons[index];
+
+                if (!icon.purpose) {
+                    continue;
+                }
+
+                const purposes = compact(icon.purpose.split(' '));
+                const purposesSet = new Set(purposes);
+
+                for (const purpose of purposesSet) {
+                    const index = purposes.indexOf(purpose);
+
+                    if (index >= 0) {
+                        purposes.splice(index, 1);
+                    }
+                }
+
+                if (purposes.length > 0) {
+                    // Is warning the right severity?
+                    context.report(resource, getMessage('iconPurposeDuplicate', context.language, [...(new Set(purposes))].join(' ')), { location, severity: Severity.warning });
```

Comment:
We might even want to lower the severity and mark it as a hint. Having a valid duplicate purpose doesn't break anything.